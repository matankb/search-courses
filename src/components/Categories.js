import React from "react";
import { connect } from "react-redux";
import { startSetCourses } from "../actions/courses";
import { topics } from "../utils/categories";

export class Categories extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            topics : topics
        }
    }

    handleOnClick = (e) => {
        const item = this.state.topics.find((topic)=> {
            return e.target.dataset.id === topic.id;
        });

        this.props.startSetCourses(item.tags);
        this.props.history.push(`/search/${item.purpose}`);
    }

    render () {
        return (
            <div className="dropdown">
               <div>Categories</div>
               <div className="dropdown__menu">
                    <ul onClick={this.handleOnClick}>
                        {this.state.topics.map((topic) => {
                            return <li key={topic.id} data-id={topic.id}>{topic.name}</li>
                        })}
                    </ul>
                </div>
            </div>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    startSetCourses: (textFilter) => dispatch(startSetCourses(textFilter))
});

export default connect(undefined, mapDispatchToProps)(Categories);

