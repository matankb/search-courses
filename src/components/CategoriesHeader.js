import React from "react";

export const CategoriesHeader = (props) => {
    return (
        <div className="search_results_page__categories">
            <div className="search-results-page__top__title">{props.title}: <span>{props.category}</span></div>
        </div>
    );
};

export default CategoriesHeader;