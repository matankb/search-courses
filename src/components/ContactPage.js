import React from "react";
import axios from "axios";
import MetaTags from 'react-meta-tags';


export class ContactPage extends React.Component {
    state = {
        name: "",
        email: "",
        text: "",
        isFormValidated: false,
        error: false,
        formSendInProgress: false,
        isMessageSent: false
    }

    onNameChange (e) {
        const name = e.target.value;
        this.setState(() => ({name}));
    }

    onEmailChange (e) {
        const email = e.target.value;
        this.setState(() => ({email}));
    }

    onTextChange (e) {
        const text = e.target.value;
        this.setState(() => ({text}));
    }

    isFormValidated () {
        const isNameFieldNotEmpty = this.state.name.trim() !== "";
        const isEmailValidated = /^.+@.+\..+$/.test(this.state.email);
        const isTextFieldNotEmpty = this.state.text.trim() !== "";

        const isFormValidated = isEmailValidated && isNameFieldNotEmpty && isTextFieldNotEmpty;
        this.setState(() => ({isFormValidated}));
    }

    handleOnSubmit (e) {
        e.preventDefault();
        this.setState(() => ({formSendInProgress: true}));

        axios.post("/api/contactus", {
            name: this.state.name,
            email: this.state.email,
            text: this.state.text
        }).then((res) => {
            this.setState(() => ({formSendInProgress: false}));
            this.setState(() => ({isMessageSent: true}));
        }).catch((err) => {
            this.setState(() => ({error: true}));
            this.setState(() => ({formSendInProgress: false}));
            console.log("Error sending email");
        });
    }

    render () {
        return (
            <div className="contact-form-container">
                <MetaTags>
                    <title>Contact Us | CourseSearch</title>
                    <meta name="description" content="Do you have any questions? Get in contact with us." />
                    <meta property="og:title" content="Contact Us | CourseSearch" />
                    <meta property="og:url" content="https://www.coursesearch.net/contact"/>
                    <meta name="twitter:title" content="Contact Us | CourseSearch"/>
                </MetaTags>
                
                <form
                    className="contact-form"
                    onKeyUp={this.isFormValidated.bind(this)}
                    onSubmit={this.handleOnSubmit.bind(this)}
                    >
                        <h2>Contact form</h2>
                        {this.state.error && <p className="error-message">Error while sending your message</p>}
                        <input
                            type="text"
                            value={this.state.name}
                            onChange={this.onNameChange.bind(this)}
                            placeholder="Your name *"
                            className="text-input contact-form__input"
                        />
                        <input
                            type="text"
                            value={this.state.email}
                            onChange={this.onEmailChange.bind(this)}
                            placeholder="Your email adress *"
                            className="text-input contact-form__input"
                        />
                        <textarea
                            type="textarea"
                            value={this.state.text}
                            onChange={this.onTextChange.bind(this)}
                            placeholder="Type your message here... *"
                            className="textarea contact-form__input"
                        />
                        <div>
                            <button className="button" type="submit" disabled={!this.state.isFormValidated || this.state.formSendInProgress || this.state.isMessageSent} >
                                {this.state.formSendInProgress ? "Sending..." : "Send"} 
                            </button>
                        </div>
                        {this.state.isMessageSent && <p className="success-message">Your message has been sent! We will get back to you as soon as possible</p>}
                </form>
            </div>
        );
    }
};

export default ContactPage;