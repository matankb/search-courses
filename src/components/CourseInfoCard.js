import React from "react";

export class CourseInfoCard  extends React.Component {

    render () {
        return (
            <div className="col col--right col--fixed"> 
                <div className="icon-container">
                    <img className="course-image" src={this.props.image} alt="course image"/>
                </div>
    
                <div className="summary-details">
                    <h2>{this.props.showPrice}</h2>
                    <a className="summary-details__link-to-course summary-details__item" href={this.props.urlToCourse} target="_blank">
                        <button className="button">
                            View course
                        </button>
                    </a>
                    <div className="summary-details__item">
                        <p className="bold">Platform:</p>
                        <p className="summary-details__item__data"> {this.props.platform}</p>
                    </div>
                    
                    {
                        !!this.props.duration &&
                        <div className="summary-details__item">
                            <p className="bold">Duration:</p>
                            <p className="summary-details__item__data"> {this.props.duration}</p>
                        </div>
                    }
                    
                    {
                        !!this.props.difficulty && 
                        <div className="summary-details__item">
                            <p className="bold">Difficulty:</p>
                            <p className="summary-details__item__data">{this.props.difficulty}</p>
                        </div>
                    }
                    
                    {
                        !!this.props.partners &&
                        <div className="summary-details__item">
                            <p className="bold">Partners</p>
                            <p className="summary-details__item__data">{this.props.partners}</p>
                        </div>
                    }
                    
                    <div className="summary-details__item">
                        <p className="bold">Certificate:</p>
                        <p className="summary-details__item__data"> {this.props.hasCertificate ? "Yes" : "No"}</p>
                    </div>
                </div>
    
            </div>
        )
    }
}