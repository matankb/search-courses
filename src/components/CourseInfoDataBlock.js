import React from "react";
import ReactMarkdown from "react-markdown";

//Re-usable component with the syllabus and description of the course.
const CourseDescription = ({description, syllabus, platform, about}) => {
    return (
        <div>
            {!!description &&
                <div className="description-block">
                    <h2 className="header-title">About this course</h2>
                    <ReactMarkdown escapeHtml={false} className="course-description" source={description} />
                </div>
            }
            
            {!!syllabus &&
                <div className="description-block">
                    <h2 className="header-title" >What you will learn</h2>
                    <ReactMarkdown escapeHtml={false} className="syllabus" source={syllabus} />
                </div>
            }
            <div className="description-block">
                <h2 className="header-title">About {platform}</h2>
                <p>{about}</p>
            </div>
        </div>
    );
}

const CourseInfoDataBlock = (props) => {
    return (
        <div className="course-info">
            
            <div className="col col--left">
                <div className="title-header-block">
                    <h1>{props.title}</h1>
                    <p>{props.shortSummary}</p>
                </div>

                <div className="show-for-desktop">
                    <CourseDescription {...props}/>
                </div>
            </div>


            <div className="col col--right">
                <div className="icon-container">
                    <img className="course-image image-full" src={props.image} alt="course image"/>
                </div>

                <div className="summary-details">
                    <h2>{props.showPrice}</h2>
                    <a className="summary-details__link-to-course summary-details__item" href={props.urlToCourse} target="_blank">
                        <button className="button">
                            View course
                        </button>
                    </a>
                    <div className="summary-details__item">
                        <p className="bold">Platform:</p>
                        <p className="summary-details__item__data"> {props.platform}</p>
                    </div>
                    
                    {
                        !!props.duration &&
                        <div className="summary-details__item">
                            <p className="bold">Duration:</p>
                            <p className="summary-details__item__data"> {props.duration}</p>
                        </div>
                    }
                    
                    {
                        !!props.difficulty && 
                        <div className="summary-details__item">
                            <p className="bold">Difficulty:</p>
                            <p className="summary-details__item__data">{props.difficulty}</p>
                        </div>
                    }
                    
                    {
                        !!props.partners &&
                        <div className="summary-details__item">
                            <p className="bold">Partners</p>
                            <p className="summary-details__item__data">{props.partners}</p>
                        </div>
                    }
                    
                    <div className="summary-details__item">
                        <p className="bold">Certificate:</p>
                        <p className="summary-details__item__data"> {props.hasCertificate ? "Yes" : "No"}</p>
                    </div>
                </div>

            </div>
            
            <div className="show-for-mobile col col--left">
                <CourseDescription {...props}/>
            </div>
        </div>
    );
};

export default CourseInfoDataBlock;

