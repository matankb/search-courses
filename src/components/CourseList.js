import React from "react";
import { connect } from "react-redux";
import CourseListItem from "./CourseListItem";
import currentPageCourses from "../selectors/currentPageCourses";
import sortBy from "../selectors/sortBy";

export class CourseList extends React.Component {
    componentDidUpdate() {
        window.scrollTo(0, 0)
    }
    
    render () {
        return (
            <div>
                <div className="course-list">
                    {this.props.currentPageCourses.map((course) =>(
                        <CourseListItem key={course.id} course={course}/>
                    ))}
                </div>
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    const sortedCourses = sortBy(state.courses.courseList, state.filters.sortBy);

    return {
        currentPageCourses: currentPageCourses(sortedCourses, state.filters.currentPage)
   }
};

export default connect(mapStateToProps)(CourseList);

