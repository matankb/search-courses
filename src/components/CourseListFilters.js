import React from "react";
import { connect } from "react-redux";

export class CourseListFilters extends React.Component {
    render () {
        return (
            <div>
                Filters
            </div>
        );
    }
}

export default CourseListFilters;