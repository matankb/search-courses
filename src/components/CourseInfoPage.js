import React from "react";
import axios from "axios";
import MetaTags from 'react-meta-tags';
import CourseInfoDataBlock from "../components/CourseInfoDataBlock";
import LoadingPage from "../components/LoadingPage";

export class CourseInfoPage extends React.Component {

    state = {
        title: "",
        image: "",
        shortSummary: "",
        description: "",
        syllabus: "",
        urlToCourse: "",
        price: "",
        duration: "",
        platform: "",
        about: "",
        difficulty: "",
        pageLoaded: false
    }

    componentDidMount () {
        this.handleGetCourse();
    }

    handleGetCourse () {
        axios.get(`/api/course/${this.props.match.params.id}`)
            .then((res) => {
                if (res.data.course === null) {
                    this.props.history.push("/");
                }
                const course = res.data.course;
  
                this.setState(() => ({
                    title: course.title,
                    image: course.image,
                    shortSummary: course.shortSummary,
                    description: course.fullDescription,
                    syllabus: course.syllabus,
                    urlToCourse: course.urlToCourse,
                    price: course.price,
                    duration: course.duration,
                    platform: course.platform,
                    about: course.aboutPlatform,
                    difficulty: course.difficulty,
                    pageLoaded: true
                }));
            })
            .catch((err) => {
                this.props.history.push("/");
            })
    }
    
    render () {
        return (
            <div>
                <MetaTags>
                    <title> {this.state.price == 0 ? "Free": ""} Online Course: {this.state.title} from {this.state.platform} | CourseSearch</title>
                    <meta name="description" content={`${this.state.shortSummary}`} />
                    <meta property="og:title" content={`${this.state.price == 0 ? "Free": ""} Online Course: ${this.state.title} from ${this.state.platform} | CourseSearch`} />
                    <meta property="og:url" content={`https://www.coursesearch.net/${this.props.match.params.id}`}/>
                    <meta name="twitter:title" content={`${this.state.price == 0 ? "Free": ""} Online Course: ${this.state.title} from ${this.state.platform} | CourseSearch`}/>
                </MetaTags>

                {!this.state.pageLoaded ? <LoadingPage /> : <CourseInfoDataBlock {...this.state} />}
            </div>
        );
    }
};

export default CourseInfoPage;
 
