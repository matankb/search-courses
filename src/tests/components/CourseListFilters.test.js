import React from "react";
import { shallow } from "enzyme";
import { CourseListFilters } from "../../components/CourseListFilters";

test("should render CourseListFilters correctly", () => {
    const wrapper = shallow(<CourseListFilters />);

    expect(wrapper).toMatchSnapshot();
});