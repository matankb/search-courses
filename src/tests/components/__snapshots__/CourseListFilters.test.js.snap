// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CourseListFilters correctly 1`] = `
<div>
  Filters
</div>
`;
